processor/src/chiplets/hasher/mod.rs

```diff
@@ -336,8 +341,8 @@ impl Hasher {
         path: &[Word],
         index: Felt,
         lookups: &mut Vec<HasherLookup>,
-    ) -> (Felt, Word, Word) {
-        let addr = self.trace.next_row_addr();
+    ) -> MerkleRootUpdate {
```

Comment:
My first approach was to change the processor. But that broke the existing constraints, so that has been undone. This is just a left over from that iteration.

The idea is to return a struct that gives names to the data, to make the code a bit more self documented. With the previous version the type doesn't really say what is the different between the first and second `Word`.

Reply:
This is generally my preferred approach as well, especially once tuples grow past 2 items

Reply:
I think it is a good improvement to name somewhat complex types (imo anything that is a tuple of 2+ components) into meaningful structs, given these will have zero-cost abstraction such as in this case.